src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -1116,4 +1101,39 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     Set(welderImageOpt, Some(jupyterImage), rstudioImageOpt).flatten
   }
 
+  private def shouldDeployWelder(cluster: Cluster): Boolean = {
+    !cluster.welderEnabled && dataprocConfig.deployWelderLabel.exists(cluster.labels.contains)
+  }
+
+  private def enableWelder(cluster: Cluster): IO[Cluster] = {
+    for {
+      _ <- IO(logger.info(s"Deploying welder on cluster ${cluster.projectNameString}"))
+      _ <- Metrics.newRelic.incrementCounterIO("welderDeployed")
+      now <- IO(Instant.now)
+      welderImage = ClusterImage(Welder, dataprocConfig.welderDockerImage, now)
+      _ <- dbRef.inTransactionIO { _.clusterQuery.enableWedler(cluster.id, ClusterImage(Welder, dataprocConfig.welderDockerImage, Instant.now)) }
+      newCluster = cluster.copy(welderEnabled = true, clusterImages = cluster.clusterImages + welderImage)
```

Comment:
will we accidentally add 2 welder images?

Reply:
It's a `Set` so I don't this so... plus this method is only called for clusters where welder is disabled.

Reply:
Actually changed this to do a DB `upsert` and `filter` on the current set to ensure we don't duplicate it.